main.c

```diff
@@ -235,8 +235,41 @@ on_check_config(TCMUService1 *interface,
 	return TRUE;
 }
 
+static gboolean
+on_mediachange(TCMUService1 *interface,
+	       GDBusMethodInvocation *invocation,
+	       guint64 size,
+	       gchar *cfgstring,
+	       gpointer user_data)
+{
+	struct tcmur_handler *handler = user_data;
+	struct tcmu_device *dev = handler->dev;
+	char *reason = NULL;
+	int ret = 0;
+
+	tcmu_set_dev_cfgstring(dev, cfgstring);
+	tcmu_update_num_lbas(dev, size);
+	tcmu_set_dev_size(dev);
+
+	tcmu_block_device(dev);
+	tcmu_flush_device(dev);
+	ret = tcmu_reopen_dev(dev);
```

Comment:
In my failover patchset I added an argument to this to control retries. I hope to merge your patch then mine over it. For that, how many retries should I set this to? Should it be infinite like it is now, or should there be a limit? What do you want the limit to be?

Reply:
There is nothing in the spec that details the amount of retries so based upon how this is written I think a few retries would be fine then if it fails to just revert the info back?